src/Swarm/TUI/Controller.hs

```diff
@@ -395,6 +395,7 @@ handleWorldEvent s (VtyEvent (V.EvKey k []))
 handleWorldEvent s (VtyEvent (V.EvKey (V.KChar 'c') [])) = do
   invalidateCacheEntry WorldCache
   continue $ s & gameState . viewCenterRule .~ VCRobot "base"
+               & gameState . focusedRobotName .~ "base"
```

Comment:
... and those two updates would be `gameState %~ setFocusedRobot "base"`

Reply:
Yes, but I think there is a better solution, since `viewCenterRule` shouldn't be use as a setter without updating `viewCenter` and `focusedRobotName` too, so probably the lens foncusing in `viewCenterRule` should be a little bit more complex than that. Something in the line of 

```haskell
viewCenterRule :: Lens' GameState ViewCenterRule
viewCenterRule = lens getter setter
  where
    getter = _viewCenterRule 
    setter gameState rule = gameState {_viewCenterRule = rule, _viewCenter = newCenterFromRule, _focusedRobotName = newRobotNameFromRule}
```

With the definition above, it'd be impossible to call `viewCenterRule .~ someRule` and forget to update the center and the robot name. This would simplify functions `modifyViewCenter`, `execConst` and `handleWorldEvent`. Again, I don't know how much of a good idea this is...

Reply:
Right, and then only declare `Getter`s for `viewCenter` and `focusedRobotName`.  Then it will be impossible for them to ever get out of sync.